extensions/agent/client/src/agentUtils.ts

```diff
@@ -0,0 +1,47 @@
+'use strict';
+
+import * as sqlops from 'sqlops';
+
+export class AgentUtils {
+
+	private static _agentService: sqlops.AgentServicesProvider;
+	private static _connectionService: sqlops.ConnectionProvider;
+	private static _fileBrowserService: sqlops.FileBrowserProvider;
+	private static _queryProvider: sqlops.QueryProvider;
+
+	public static async getAgentService(): Promise<sqlops.AgentServicesProvider> {
+		if (!AgentUtils._agentService) {
+			let currentConnection = await sqlops.connection.getCurrentConnection();
+			this._agentService = sqlops.dataprotocol.getProvider<sqlops.AgentServicesProvider>(currentConnection.providerName, sqlops.DataProviderType.AgentServicesProvider);
+		}
+		return AgentUtils._agentService;
+	}
+
+	public static async getDatabases(ownerUri: string): Promise<string[]> {
+		if (!AgentUtils._connectionService) {
+			let currentConnection = await sqlops.connection.getCurrentConnection();
+			this._connectionService = sqlops.dataprotocol.getProvider<sqlops.ConnectionProvider>(currentConnection.providerName, sqlops.DataProviderType.ConnectionProvider);
```

Comment:
why is this method caching the connection service into a static variable?  It doesn't seem to try to read the cached value, so it seems like saving into a local variable removes a side-effect from this method.